File: itzel_be/src/main/java/com/itzel/servicio/contabilidad/EstructuraService.java
package com.itzel.servicio.contabilidad;

import com.itzel.modelo.contabilidad.Estructura;
import com.itzel.repositorio.contabilidad.EstructuraR;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class EstructuraService {
    @Autowired
    private EstructuraR dao;

    public List<Estructura> findAll(){
        return dao.findAll();
    }
    // Devuelve la entidad con el nivel más alto
    public Estructura findTopByOrderByNivelDesc(){
        return dao.findTopByOrderByNivelDesc();
    }
    // Devuelve todas las entidades con un nivel específico
    public List<Estructura> findByNivel(Integer nivel){
        return dao.findByNivel(nivel);
    }
}

File: itzel2.1/src/app/componentes/servicios/contabilidad/estructura.service.ts
import { Injectable } from '@angular/core';
import { environment } from '../../../../environments/environment';
import { HttpClient } from '@angular/common/http';
import { Estructura } from '../../modelos/contabilidad/estructura.model';
import { Observable } from 'rxjs';
const apiUrl = environment.API_URL;
const baseUrl = `${apiUrl}/estructura`;
@Injectable({
  providedIn: 'root',
})
export class EstructuraService {
  constructor(private http: HttpClient) {}
  // Devuelve todas las estructuras
  estructuraGetAll() {
    return this.http.get<Estructura>(`${baseUrl}`);
  }
  // Devuelve la entidad con el nivel más alto
  findTopByOrderByNivelDesc() {
    return this.http.get<Estructura>(`${baseUrl}/maxnivel`);
  }
  // Devuelve todas las entidades con un nivel específico
  findByNivel(nivel: number) {
    return this.http.get<Estructura>(`${baseUrl}/findByNivel?nivel=${nivel}`);
  }
}
